controllers/provider-alicloud/pkg/controller/worker/machines.go

```diff
@@ -40,13 +44,94 @@ func (w *workerDelegate) MachineClassList() runtime.Object {
 	return &machinev1alpha1.AlicloudMachineClassList{}
 }
 
+// getSeedCloudProviderCredentials gets Seed's cloud provider's credentials from BackupBucket SecretRef
+func (w *workerDelegate) getSeedCloudProviderCredentials(ctx context.Context) (*alicloud.Credentials, error) {
```

Comment:
Why is all this logic here, in the worker delegate? The `Worker` control loops should only manage the worker machines. 
I consider what you are trying to achieve (image sharing) more to be related to infrastructural preparations, so please move your code to your `Infrastructure` control loop.

Reply:
done. Previously we just wanted to share the image when actually deploying the worker. But it indeed make better sense to consider it as a part of infrastructural preparations. Thanks for the advice :)